File: Aula5/declaraClasse.java
import java.lang.reflect.Field;

/**
 * @author : Guilherme Wachs Lopes (gwachs@fei.edu.br)
 * @file : reflection
 * @created : quinta mar 04, 2021 21:16:43 -03
 */

class Teste {
  int a;
  int b;
  int c;

  public int getA() {
    return a;
  }

  public void setA(int a) {
    this.a = a;
  }
}

public class reflection {
  public static void main(String[] args) {
    Teste a = new Teste();
    a.setA(20);

    for (Field f : Teste.class.getDeclaredFields()) {
      System.out.println(f.getName());
    }
  }
}

File: Aula5/declaraClasse.cpp
/**
 * @author      : Guilherme Wachs Lopes (gwachs@fei.edu.br)
 * @file        : declaraClasse
 * @created     : quinta mar 04, 2021 21:54:18 -03
 */

class Teste {
  private:
    int a;
  public:
    float b;
  protected:
    double t;
};

class TesteSub : private Teste {
  void teste(){
    this->t = 9;
  }
};

class TesteSubSub : public TesteSub {
  void teste2(){
  }
};



int main() {
  
 Teste x;
 TesteSub y;
  y.
 


  return 0;
}
